### PR TITLE
[hw,i2c,dv] Add waiver file for host mode FSM transitions

### PR DESCRIPTION
**Pull request by crteja**
- Added FSM coverage waiver file based on https://github.com/lowRISC/opentitan/issues/18651
- Added override for VCS coverage config to disable coverage analysis of `i2c_protocol_cov` as this is a testbench module

### CODE REVIEW

File: hw/ip/i2c/dv/cov/i2c_cov_excl.el
Comment (jdonjdon) [inline hw/ip/i2c/dv/cov/i2c_cov_excl.el:16]: Please add annotation like this
https://github.com/lowRISC/opentitan/blob/c6d13942ec3df7be63ff6ab172f59823d3e9be3c/hw/ip/keymgr/dv/cov/keymgr_cov_excl.el#L5

Comment (crteja) [reply hw/ip/i2c/dv/cov/i2c_cov_excl.el:16]: Added annotation in [aaa5c0a](https://github.com/lowRISC/opentitan/compare/4af57a4c4b2049986260996cdabe3bd5ce0829c4..aaa5c0a1964d4310465c5a9a97fc55a4e65a23aa)

Comment (jdonjdon) [reply hw/ip/i2c/dv/cov/i2c_cov_excl.el:16]: awesome! thanks.

**Review by jdonjdon — APPROVED** [review]

**Review by marnovandermaas — COMMENTED** [review]
Before merging this we need to add `i2c_protocol_cov` FSM as well.

**Comment by crteja** [comment]
> Before merging this we need to add `i2c_protocol_cov` FSM as well.

`i2c_protocol_cov` is excluded using VCS coverege config file override in `hw/ip/i2c/dv/cov/cov_excl.cfg`. This is a new file added in the PR.

**Comment by marnovandermaas** [comment]
> > Before merging this we need to add `i2c_protocol_cov` FSM as well.
> 
> `i2c_protocol_cov` is excluded using VCS coverege config file override in `hw/ip/i2c/dv/cov/cov_excl.cfg`. This is a new file added in the PR.

Ah yes, sorry for overlooking that.

**Review by marnovandermaas — APPROVED** [review]
Looks good to me, could you add @jdonjdon as co-author for this commit? I have not tested this by the way, @crteja can you confirm that you have tested this and everything looks as you expect in the coverage report?

**Comment by crteja** [comment]
> Looks good to me, could you add @jdonjdon as co-author for this commit? I have not tested this by the way, @crteja can you confirm that you have tested this and everything looks as you expect in the coverage report?

Added @jdonjdon as Co-Author.
Yes, I have run a sanity test locally with coverage enabled. `i2c_protocol_cov` is not visible in coverage report and `i2c_fsm.sv` transition are excluded

**Comment by marnovandermaas** [comment]
Thanks @crteja! Let's wait for CI to pass and then we can merge this one.

**Pull request merged**
